### DISCUSSION TITLE
Issues with FixClient versions and questions custom fields

### DISCUSSION THREAD

**Discussion by Artemin** [GaryHughes/FixClient#13] [Q&A] [answered]
Hi Gary

I have tried to add custom fields on the 5.0 release version of FixClient, but encountered a error message as below
<img width="337" alt="image" src="https://github.com/GaryHughes/FixClient/assets/44104482/63404c85-a14b-4172-9028-ebed19802fdc">

Then I cloned your repository and generate another FixClient v5.0 using VS. That version worked. Probably the release version needs an update?

---
I am also encountering issues with the custom fields feature page.
- What is the GATE drop-down box used for? 
- Upon adding custom fields, how to apply them to any of the existing messages? 
- By any chance to add new customer FIX messages on this tool? 

Appreciate your response.

**Comment by GaryHughes** [comment, marked as answer]
Hi @Artemin you are correct I need to do another release as that custom field exception has been fixed. I was waiting to complete some other changes but I'll do the release sooner.

Regarding the GATE dropdown, I originally built this program when working for a company (now acquired by Virtu who open sourced FixClient and where you will see this is forked from) that had an application called GATE that routed FIX traffic amongst other things. This button automatically added several commonly used custom tags for use with GATE. There are still people at Virtu that use FIX Client so I leave the functionality in place.

Custom field definitions are used to populate the descriptions/logs etc when messages with these fields are received via a session connection, pasted into the message editor, or parsed from a log file in the parser view. 

Custom messages are possible and some hard coded custom messages are supported right now. If you look at [leicographer.py](https://github.com/GaryHughes/FixClient/blob/master/Fix.Lexicographer/lexicographer.py) you will see some messages being injected into the data dictionary. Supporting this dynamically in the UI is possible but would be a reasonably large amount of work, if you really want this please open an issue.

Thanks for the questions, I know this could really use some documentation so please let me know of anything that isn't clear or you can't figure out.

**Reply by Artemin** [reply]
Thanks you for the prompt reply. It is very clear now.

**Comment by ymanudhane** [comment]
Hi Gary, Is there any file which can help to understand each and every session setting providing its function? What is significance of Bind host and Bind port vs Host and Port?

**Reply by GaryHughes** [reply]
There isn't at the moment but I will document these settings for you.
